coco_tools/split.py

```diff
@@ -1,2 +1,142 @@
-def split():
-    print("Unimplemented!")
+import json
+import pandas as pd
+import numpy as np
+from pathlib import Path
+from coco_tools.error import COCOToolsError
+
+
+def split(dataset_path, ratio, names):
+    """Splits the dataset into multiple parts based on the given ratio.
+
+    Within the dataset, one image can have multiple annotations. `split` splits
+    the dataset by the number of images, and splits the annotations based on the
+    images that they belong to.
+
+    For example, in a dataset of 1000 images, a ratio of `70:20:10` would split
+    the dataset into three datasets containing `700`, `200` and `100`
+    respectively.
+    """
+
+    # Extract and validate the inputs.
+    dataset_path = Path(dataset_path)
+    ratio = __extract_ratio(ratio)
+    names = __extract_names(names)
+
+    # Some additional input validation.
+    if len(ratio) != len(names):
+        raise COCOToolsError("ratio and names should be of same length")
+
+    # Load the dataset from `dataset_path`.
+    raw_data = None
+    try:
```

Comment:
Probably can do this neater

Reply:
```
    # Load the dataset from `dataset_path`.                                                          
    try:
        with open(str(dataset_path), "r") as dataset_file:
            raw_data = json.load(dataset_file)

            # Extract `images` and `annotations`self.                                                
            images = raw_data.pop("images")
            annotations = raw_data.pop("annotations")

            # Initialize the new datas.                                                              
            new_datas = [raw_data.copy() for _ in ratio]

            # Split the data.                                                                        
            __split_data(new_datas, ratio, images, annotations)

            # Output the results to the corresponding files.                                         
            for (i, new_data) in enumerate(new_datas):
                with open(__derive_path(dataset_path, names[i]), "w") as output_file:
                    json.dump(new_data, output_file)
    except FileNotFoundError:
        raise COCOToolsError(f"file \"{dataset_path}\" not found")
```

Not much better though

Reply:
Hmm, I think the more Go style error handling with the early return is better.